ui/oopsyraidsy/data/07-dt/ultimate/futures_rewritten.ts

```diff
@@ -1,18 +1,152 @@
+    'FRU Explosion': '9D1B', // P2 Explosion from Holy Light
+    'FRU Hiemal Storm': '9D40', // P2.5 Crystal of Light puddle
+    'FRU Sinbound Blizzard III': '9D46', // P2.5 Crystal of Darkness cone
+    'FRU Dark Blizzard III': '9D57', // P3, P4 Dark Blizzard III
+    'FRU Sinbound Meltdown 1': '9D64', // P3 Hourglass beam (after 1st hit)
+    'FRU Apocalypse': '9D69', // P3 Apocalypse
+    'FRU Akh Rhai': '9D2E', // P4 Akh Rhai
+    'FRU Hallowed Wings 1': '9D23', // P4 Dragonsong Hallowed Wings
+    'FRU Hallowed Wings 2': '9D24', // P4 Dragonsong Hallowed Wings
+    'FRU Maelstrom': '9D6B', // P4 Crystallize Time hourglass
+    'FRU Tidal Light 1': '9D3C', // P4 Crystallize Time exaflare
+    'FRU Tidal Light 2': '9D3D', // P4 Crystallize Time exaflare
+    'FRU Path of Light': '9D74', // P5 Fulgent Blade exaflare
+    'FRU Path of Darkness': '9D75', // P5 Fulgent Blade exaflare
+    'FRU Cruel Path of Light': '9CB7', // P5 Polarizing Paths follow-up
+    'FRU Cruel Path of Darkness': '9CB8', // P5 Polarizing Paths follow-up
+  },
+  damageFail: {
+    'FRU Unmitigated Explosion 1': '9CC4', // P1 tower fail
+    'FRU Unmitigated Explosion 2': '9D81', // P5 tower fail
+    'FRU Refulgent Fate': '9D17', // P2, P4 tether break
+  },
+  gainsEffectWarn: {
+    'FRU Damage Down': 'B5F',
+    'FRU Bleeding': 'C05', // standing in the puddle.
+    'FRU Doom': '9D4',
+  },
+  gainsEffectFail: {
+    'FRU Mark of Mortality': '1114', // stack fail debuff
+  },
+  shareWarn: {
+    'FRU Sinsmite': '9CD5', // P1 spread during Cyclonic Break
+    'FRU Sinbound Thunder III': '9CE0', // P1 spread during Utopian Sky
+    'FRU Bow Shock': '9CCF', // P1 Bow Shock (tethered thunder)
+    'FRU House of Light 1': '9D0E', // P2 House of Light during Diamond Dust, Mirror Mirror
+    'FRU Banish III Divided': '9D1F', // P2 Banish III Divided (spread)
+    'FRU Sinbound Meltdown 2': '9D2B', // P3 Hourglass beam 1st hit
+    'FRU Dark Fire III': '9D54', // P3 Dark Fire III
+    'FRU Dark Eruption': '9D52', // P3, P4 Dark Eruption
+    'FRU Spirit Taker': '9D61', // P3, P4 Spirit Taker
+    'FRU Longing of the Lost': '9D31', // P4 Crystallize Time dragon explosion
+  },
+  shareFail: {
+    'FRU House of Light 2': '9CFC', // P2 House of Light after Light Rampant, P4 House of Light
+    'FRU Darkest Dance': '9CF6', // P3 Darkest Dance tank buster
+    'FRU Somber Dance 1': '9D5C', // P4 Somber Dance 1st
+    'FRU Somber Dance 2': '9D5D', // P4 Somber Dance 2nd
+    'FRU Wings Dark and Light 1': '9D7A', // P5 Wings Dark and Light cleave
+    'FRU Wings Dark and Light 2': '9D7B', // P5 Wings Dark and Light cleave
+    'FRU Wings Dark and Light 3': '9BC7', // P5 Wings Dark and Light tether
+    'FRU Wings Dark and Light 4': '9BC8', // P5 Wings Dark and Light tether
+  },
+  soloWarn: {
+    'FRU Sinsmoke': '9CD3', // P1 stack during Cyclonic Break
+    'FRU Banish III': '9D1E', // P2 Banish III (stack)
+  },
```

Comment:
(sorry for the massive line selection on this comment)

Does it make sense to update these entries to be more descriptive? For example:

```ts
  damageWarn: {
    // ...
    'FRU P1 Burnt Strike Fire Boss': '9CC1', // P1 Burnt Strike (fire, by Fatebreaker)
    'FRU P1 Burnt Strike Thunder Boss': '9CC5', // P1 Burnt Strike (thunder, by Fatebreaker)
    'FRU P1 Burnt Strike Fire Clone': '9CE1', // P1 Burnt Strike (fire, by Fatebreaker's Image)
    'FRU P1 Burnt Strike Thunder Clone': '9CE3', // P1 Burnt Strike (thunder, by Fatebreaker's Image)
```